base/interactiveutil.jl

```diff
@@ -229,46 +233,15 @@ typesof(args...) = Tuple{map(a->(isa(a,Type) ? Type{a} : typeof(a)), args)...}
 
 gen_call_with_extracted_types(fcn, ex0::Symbol) = Expr(:call, fcn, Meta.quot(ex0))
 function gen_call_with_extracted_types(fcn, ex0)
-    if isa(ex0, Expr) &&
-        any(a->(Meta.isexpr(a, :kw) || Meta.isexpr(a, :parameters)), ex0.args)
-        # keyword args not used in dispatch, so just remove them
-        args = filter(a->!(Meta.isexpr(a, :kw) || Meta.isexpr(a, :parameters)), ex0.args)
-        return Expr(:call, fcn, esc(args[1]),
-                    Expr(:call, :typesof, map(esc, args[2:end])...))
-    end
+    exret = Expr(:call, :error, "expression is not a function call or symbol")
+
     ex = expand(ex0)
     if isa(ex, Expr) && ex.head == :call
-        return Expr(:call, fcn, esc(ex.args[1]),
+        exret = Expr(:call, fcn, esc(ex.args[1]),
                     Expr(:call, :typesof, map(esc, ex.args[2:end])...))
     end
-    exret = Expr(:call, :error, "expression is not a function call or symbol")
-    if !isa(ex, Expr)
-        # do nothing -> error
-    elseif ex.head == :call
```

Comment:
this does look apparently redundant, but the `:body` and `:thunk` branches seem like they might still matter